synapse/models/inet.py

```diff
@@ -35,6 +35,10 @@
 
 rfc6598 = ipaddress.IPv4Network('100.64.0.0/10')
 
+# defined from https://x.com/4A4133/status/1887269972545839559
+ja4_regex = r'^([tqd])(\d{2})([di])(\d{2})(\d{2})([a-z0-9]{2})_([0-9a-f]{12})_([0-9a-f]{12})$'
```

Comment:
I think the second group needs to be `[sd\d]` as SSL/DTLS have s/d as the first char

Reply:
The spec in the their github repo at https://github.com/FoxIO-LLC/ja4/blob/main/technical_details/JA4.md lists the options as
```
0x0304 = TLS 1.3 = “13”
0x0303 = TLS 1.2 = “12”
0x0302 = TLS 1.1 = “11”
0x0301 = TLS 1.0 = “10”
0x0300 = SSL 3.0 = “s3”
0x0002 = SSL 2.0 = “s2”
0xfeff = DTLS 1.0 = "d1"
0xfefd = DTLS 1.2 = "d2"
0xfefc = DTLS 1.3 = "d3"

Unknown = “00”